thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
To create non static inner class, we need to do like following sample code.

```java
public class Test {

    public class AInnerClasss {
    }


    public static void main(String arg[]) {
        Test parent = new Test();
        AInnerClasss child =  parent.new AInnerClasss();
        // We can not create AInnerClasss instance like following.
        // AInnerClasss child = new AInnerClasss()
    }
}
```

To create `AInnerClass` instance, we need instance of parent class. In this case, we require instance of `Test`.

I wonder we can create or not`AInnerClass` instance in gson adapter because we may not create instance of parent class.

Can we break the rule by using reflection? I don't know about it.